src/database/search_handler.cc

```diff
@@ -201,18 +201,18 @@ std::shared_ptr<ASTNode> SearchParser::parseSearchExpression()
                     throw_std_runtime_error("Cannot construct ASTAndOperator without lhs");
                 if (!expressionNode)
                     throw_std_runtime_error("Cannot construct ASTAndOperator without rhs");
-                std::shared_ptr<ASTNode> lhs(nodeStack.top());
+                auto lhs = std::move(nodeStack.top());
                 nodeStack.pop();
-                nodeStack.push(std::make_shared<ASTAndOperator>(sqlEmitter, lhs, expressionNode));
+                nodeStack.push(std::make_shared<ASTAndOperator>(sqlEmitter, std::move(lhs), std::move(expressionNode)));
```

Comment:
The change would be more obvious if you move the declaration of `expressionNode` inside the while loop.

Reply:
Done.